src/Storage/Device/CloudflareR2.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Utopia\Storage\Device;
+
+use Utopia\Storage\Storage;
+
+class CloudflareR2 extends S3
+{
+    /**
+     * Regions constants
+     */
+
+    /**
+     * Regions constants
+     */
+    const WNAM = 'us-west-1';
+
+    const ENAM = 'us-east-1';
+
+    const WEUR = 'eu-west-1';
+
+    const EEUR = 'eu-east-1';
+
+    const APAC = 'ap-southeast-1';
+
+    const AUTO = 'auto';
+
+    /**
+     * CloudflareR2 Constructor
+     *
+     * @param  string  $root
+     * @param  string  $accessKey
+     * @param  string  $secretKey
+     * @param  string  $bucket
+     * @param  string  $region
+     * @param  string  $acl
+     */
+    public function __construct(string $root, string $accessKey, string $secretKey, string $bucket, string $region = self::APAC, string $acl = self::ACL_PRIVATE)
+    {
+        parent::__construct($root, $accessKey, $secretKey, $bucket, $region, $acl);
+        $this->headers['host'] = $bucket.'.'.'s3'.'.'.$region.'.cloudflarestorage.com';
```

Comment:
Do you have any documentation on the host?

Reply:
An S3 API URL looks like this:
https://accountid.r2.cloudflarestorage.com/bucketname

Where account id is your CF Account ID, like this:
https://905fa3fe223dcff4bf6b46911342bcfd.r2.cloudflarestorage.com/test

If you are using the EU Juristiction, you need to add .eu before the .r2, like this:
https://905fa3fe223dcff4bf6b46911342bcfd.eu.r2.cloudflarestorage.com/test